### ISSUE TITLE
Some troubles and missing Infos - How to choose Landing Destination, Target Orbit etc?

### ISSUE THREAD

**Issue by Caliban2017** [pmborg/SpaceX-RO-Falcons#3]
Hi, first of all, great Work on this Mod! I tried the Demo Mission a little bit but it makes some troubles. All worked well so far, Boostback burn worked, Booster tried to Land on the LZ1 but then the Landing Platform seems to fall through the Ground and landet into Water underneath it. So the Landing of course failed. I tried to launch another Mission with a Drone Ship but here i got some Questions. Where can i change the Target for the Mission? I'm not familiar with kos and that Scripts. I can only select the Target ( pressing 0 and 3) but where can i define another Target Orbit or another Landing Zone like the Drone Ships?

I also had some trouble loading the save games. It throw some Errors on loading the Rockets with missing Parts. I used a clean install of KSP v1.12.3.

Some more informations for this would be nice. 

Greetings from Germany and sorry for my bad english.

**Comment by pmborg** [comment]
Hello, all questions make sense:

[1] "**but then the Landing Platform seems to fall through the ground and land in the water underneath it. So the Landing of course failed.**"
- This is related to a "KSP bug" where the land is not correctly loaded so the workaround
is to switch to another vessel (using the switch keys) and come back until it loaded the terrain correctly as can be seen in this video:
https://youtu.be/NK_UHjtmeV8?t=419
From second **419** to until second **456**
https://youtu.be/NK_UHjtmeV8?t=456

[2] "**I tried to launch another Mission with a Drone Ship but here I got some Questions. Where can I change the Target for the Mission?**"
- All the missions are pre-defined and pre-configured, the boat (in case of a water landing) needs to be in the expected position, then the falcon-ST1 can correct about +-3km from that position.
each mission has its load and speed so the drone needs to be in a pre-calculated/expected position.

[3] "**I'm not familiar with kos and Scripts. I can only select the Target ( pressing 0 and 3) but where can i define another Target Orbit or another Landing Zone like the Drone Ships?**"
- You have this file where you can choose between land on LAND or on SHIP:
-**[KSP DIRECTORY]\Ships\Script\boot\USER-CONFIG.KS**
But that depends on the mission, some missions allow that choice and some missions only allow the ocean landing.

[4] "**I also had some trouble loading the save games. It throws some Errors on loading the Rockets with missing Parts. I used a clean install of KSP v1.12.3.**"
- There are old SPACEX parts that are considered as outdated, so I just created an updated version with another name
so load all craft files started by **PMBT....**
And you will have there all SPACEX vehicles already set up and configured based on old crafts and missions.

**Comment by Caliban2017** [comment]
Thank you for your explanation, i tried it again, but it sadly make some more troubles, when i choose the Sea Landing. The first Question is here again, how does i choose the target Drone Ship? Or is it depending on the used Rocket? I see only LAND and SEA in the Config File. I tried now the Sea landing with the Demo Mission and it says it will land on JRTI but here are some strange things happening, after the Stage Sep, the upper Stage shoot out a Parachute which is attached to a Fairing Half. On the Boost Back burn everything works fine so far but the Booster misses the Target for about ~25km and land in Water. Shortly  after the entry Burn, the booster begin to spin around, maybe it tries to correct the distance to the Drone Ship? It spins fast, ending up in horizontal position and it boost again in direction of the Drone Ship, but like i said, it is not enough, it misses the target and land into Water. And the second Stage seems to cut off the Engine and does not complete the Orbit Burn. It just fell back into the Atmosphere and splash down in the Ocean. 

This is really a very cool Demo and give some nice SpaceX feeling in KSP, it is very nice work but for my oppinion it is not very usefull because it is a bit buggy and unflexible, except you know how to code your own flight profiles. Here would be a advanced Script cool, where you can enter your desired target orbit(height) in the kos console.

**Comment by pmborg** [comment]
**"The first Question is here again, how does i choose the target Drone Ship? Or is it depending on the used Rocket? "**

- All falcon missions are coded in real life and here is about the same thing, the main missions were coded as supposed to be, there isn't a magic button of a setting its just pure code, and yes it depends on the mission and of the falcon version:
(here for example is where the target is selected at the end of the code)
https://github.com/pmborg/SpaceX-RO-Falcons/blob/main/Ships/Script/boot/stage-main.ks

- But who knows if one of these days I will have time to come back and save a video again,
for all types of missions in a near future.

**"I see only LAND and SEA in the Config File. I tried now the Sea landing with the Demo Mission and it says it will land on JRTI but here are some strange things happening"**

- I have no idea what missions are you testing.
Is this Mission?: (without cape right?)
KSP1-10-11-12-DEMO-BASE-DroneShip1-2-LZ1-2 (defautl Falcon 9 ?)

**"This is really a very cool Demo and give some nice SpaceX feeling in KSP, it is very nice work but for my oppinion it is not very usefull because it is a bit buggy and unflexible, except you know how to code your own flight profiles. Here would be a advanced Script cool, where you can enter your desired target orbit(height) in the kos console."**

- Here all work perfect for me maybe you are doing something that is not supposed...
I don't know what is that.
In order to get an idea of what is going on I need:

- If you provide a ZIP with this content, maybe I can figure out of is going on:
KSP.log
Ships/Script/boot/*

**Comment by Caliban2017** [comment]
Thanks for the explanation, that is what i thought. It is a really cool demo but for a regular savegame it is of course useless, when u have to edit the scripts. In that case it is easier to use MechJeb Landing Guidance. The only problem here is sometimes the timing to reach the orbit the same time and still land the booster.

I do not found any Missions for the Falcon 9. I used in that case the Savegame for the Career Mode (InterStar). I used the Crew Dragon Falcon 9 for this test flight.

But i can of course test it again and save some log files, if it will fail again...
